### DISCUSSION TITLE
C wrapper for CacheLib APIs

### DISCUSSION THREAD

**Discussion by guptask** [facebook/CacheLib#79] [Q&A] [answered]
For integration with a benchmark, a C wrapper is needed for the CacheLib APIs. Is there any existing C wrapper for CacheLib that I refer to ?

**Comment by agordon** [comment, marked as answer]
Hello @guptask ,

Currently there is no C wrapper, but since we are working on wrappers for other programming languages, let me see if I can come up with some ad-hoc C solution. Can you describe a bit more the benchmark you are trying to integrate?
Which functionality of CacheLib are you interested in?

**Reply by guptask** [reply]
I'm interested in building a fio engine for cachelib.

**Comment by sathyaphoenix** [comment]
CacheLib APIs rely on RAII semantics for the ItemHandle. This might be a big source of hurdle for building an efficient C wrapper.

**Comment by pbhandar2** [comment]
Closing this discussion as it has been inactive. If this is still relevant, then please create a new discussion post and we will get back to you as soon as we can. Thank you!
